sdk/nodejs/resource.ts

```diff
@@ -174,6 +233,27 @@ export abstract class Resource {
         }
         this.__protect = !!opts.protect;
 
+        // Collapse any `Alias`es down to URNs. We have to wait until this point to do so because we do not know the
+        // default `name` and `type` to apply until we are inside the resource constructor.
+        this.__aliases = [];
+        if (opts.aliases) {
+            for (const alias of opts.aliases) {
+                this.__aliases.push(output(alias).apply(a => {
+                    if (typeof a === "string") {
+                        return output(a);
+                    } else {
+                        return createUrn(
+                            output(a.name).apply(n => n || name),
+                            output(a.type).apply(ty => ty || t),
+                            a.parent || opts.parent,
```

Comment:
terrifying.  but i guess it's ok since 'output' does not rip apart resources?

Reply:
Although I agree this is overall rather terrifying - is there anything in particular you are concerned about here?